core/src/main/java/info/nightscout/androidaps/utils/extensions/UserEntryExt.kt

```diff
@@ -1,224 +1,230 @@
-        UserEntry.Action.NS_PAUSED                  -> R.color.defaulttext
-        UserEntry.Action.NS_QUEUE_CLEARED           -> R.color.defaulttext
-        UserEntry.Action.NS_SETTINGS_COPIED         -> R.color.defaulttext
-        UserEntry.Action.ERROR_DIALOG_OK            -> R.color.defaulttext
-        UserEntry.Action.ERROR_DIALOG_MUTE          -> R.color.defaulttext
-        UserEntry.Action.ERROR_DIALOG_MUTE_5MIN     -> R.color.defaulttext
-        UserEntry.Action.OBJECTIVE_STARTED          -> R.color.defaulttext
-        UserEntry.Action.OBJECTIVE_UNSTARTED        -> R.color.defaulttext
-        UserEntry.Action.OBJECTIVES_SKIPPED         -> R.color.defaulttext
-        UserEntry.Action.STAT_RESET                 -> R.color.defaulttext
-        UserEntry.Action.DELETE_LOGS                -> R.color.defaulttext
-        UserEntry.Action.DELETE_FUTURE_TREATMENTS   -> R.color.defaulttext
-        UserEntry.Action.EXPORT_SETTINGS            -> R.color.defaulttext
-        UserEntry.Action.IMPORT_SETTINGS            -> R.color.defaulttext
-        UserEntry.Action.RESET_DATABASES            -> R.color.defaulttext
-        UserEntry.Action.EXPORT_DATABASES           -> R.color.defaulttext
-        UserEntry.Action.IMPORT_DATABASES           -> R.color.defaulttext
-        UserEntry.Action.OTP_EXPORT                 -> R.color.defaulttext
-        UserEntry.Action.OTP_RESET                  -> R.color.defaulttext
-        UserEntry.Action.SMS_BASAL                  -> R.color.basal
-        UserEntry.Action.SMS_BOLUS                  -> R.color.iob
-        UserEntry.Action.SMS_CAL                    -> R.color.calibration
-        UserEntry.Action.SMS_CARBS                  -> R.color.carbs
-        UserEntry.Action.SMS_EXTENDED_BOLUS         -> R.color.extendedBolus
-        UserEntry.Action.SMS_LOOP_DISABLED          -> R.color.loopDisabled
-        UserEntry.Action.SMS_LOOP_ENABLED           -> R.color.loopClosed
-        UserEntry.Action.SMS_LOOP_RESUME            -> R.color.loopClosed
-        UserEntry.Action.SMS_LOOP_SUSPEND           -> R.color.loopSuspended
-        UserEntry.Action.SMS_PROFILE                -> R.color.white
-        UserEntry.Action.SMS_PUMP_CONNECT           -> R.color.loopDisconnected
-        UserEntry.Action.SMS_PUMP_DISCONNECT        -> R.color.loopDisconnected
-        UserEntry.Action.SMS_SMS                    -> R.color.defaulttext
-        UserEntry.Action.SMS_TT                     -> R.color.tempTargetConfirmation
-        UserEntry.Action.TT_DELETED_FROM_NS         -> R.color.tempTargetConfirmation
-        UserEntry.Action.TT_FROM_NS                 -> R.color.tempTargetConfirmation
-        UserEntry.Action.TT_CANCELED_FROM_NS        -> R.color.tempTargetConfirmation
-        else                                        -> R.color.defaulttext
+        Action.EXTENDED_CARBS             -> R.color.carbs
+        Action.TEMP_BASAL                 -> R.color.basal
```

Comment:
feel free to create new constant R.color for every group

Reply:
I didn't want to create new colors, I think we should try to limit and standardize the existing colors to make it easier to understand (ex Orange Carb, Blue Insulin...) and use the same colors everywhere for the same theme (Text, icons, graphs...).
But for "multicolor items" (ex Careportal, Loop, BG values), or when there is no standard color (Pump) we could propose new ones or use the most frequently use (if not already taken by another theme) as "theme color".
When 2 themes have colors that are too close (eg TT, BG In Range and Closed Loop), I don't know which is the best solution ...

Reply:
If you Use Colors you Can have a look into my dark/light Branch.
To will be hopefullly merged in the next times.
I began to refactor colors and the only access to the colors succed by Attributes.i began to rename Colors to gray, Light gray ... and so on

Reply:
Good idea to work on colors, and reduce / harmonize the diversity.
I think it's better if we could "name" colors according to standard use in AAPS (I prefer "Light Carb" and "Dark Carb" than neutrals "Light Orange" / "Dark Orange")
In this PR, I select colorId with R.color.xxxx (I don't know how it works with Materal, I suppose we can replace Id by an attribute in UserEntryExt.kt and replace resourceHelper.gc in TreatmentsUserEntryFragment...)
I'll take a look in your PR.

Reply:
The Standard use will be clear by the Attributes and Not by the colorname in the Future .
In my branch resourceHelper.gc will Not be used anymore.
Instead we use the new function resourceHelper.getColorbyAttributes

Reply:
For UserEntries, Milos asked me to create "groups of Color" according to actions. I initialize 8 groups.
Do you think we could map these groups (we can of course update this list) with your attributes?

Reply:
I think yes.
But the access function is only available in my Branch in the Moment .
Its nö Time to merge to avoid additional work.

Reply:
Don't bother, I'll take a look on my side too with your PR, to see the impact (and prepare modifications if your PR is merged before mine).

Reply:
Perfect. And don't Hesitate for question of help . Still Write me a Mail and i help wherever i Can.